pages/index.tsx

```diff
@@ -1,19 +1,25 @@
 import type { NextPage } from "next";
-import useSwr from "swr";
+import { useEffect, useState } from "react";
 import PackageList from "../components/PackageList";
 
-const fetcher = (url: string) => fetch(url).then((res) => res.json())
-
 const Home: NextPage = () => {
+  const [data, setData] = useState(null);
+  const [isLoading, setLoading] = useState(false);
 
-  const { data, error } = useSwr("/api/package-reader",
-    fetcher
-  )
+  useEffect(() => {
+    setLoading(true);
+    fetch("/api/package-reader")
+      .then((data) => data.json())
+      .then((data) => {
+        setData(data);
+        setLoading(false);
+      });
+  }, []);
 
-  if (error) return <div>Failed to load package names</div>
-  if (!data) return <div>Loading...</div>
+  if (isLoading) return <div>Loading...</div>;
+  if (!data) return <div>NO data found!</div>;
```

Comment:
with the above modifications, this check should then become `if (data.length === 0) ...`